src/utils/faqGenerator.ts

```diff
@@ -0,0 +1,47 @@
+import fs from "fs";
+import path from "path";
+import matter from "gray-matter";
+
+type FAQProps = {
+  title: string;
+  date: string;
+  link: string;
+};
+
+const MDX_EXTENSION = ".mdx";
+const POST_DIR = "faqs";
+const ROOT_DIR = "./";
+
+const faqGenerator = () => {
+  const getFAQs = () => {
+    const files = fs.readdirSync(path.join(ROOT_DIR, POST_DIR));
+
+    if (!files.length) {
+      return null;
+    }
+
+    const faqs = files.map((file) => {
```

Comment:
```ts
   const mdxFiles = files.filter(
      (file) => path.extname(file) === MDX_EXTENSION
    );
    const faqs = mdxFiles.map((file) => {
      const source = fs.readFileSync(path.join(ROOT_DIR, POST_DIR, file), {
        encoding: "utf-8",
      });

      const { data } = matter(source);

      const faqList = data as FAQProps;
      return {
        ...faqList,
        link: file.replace(MDX_EXTENSION, ""),
      };
    });
```
This provides better typesupport and is more clear.

Reply:
Totally, super vai.